jade/jobs/job_configuration.py

```diff
@@ -207,6 +211,73 @@ def check_job_dependencies(self, submitter_params):
                 "Submitting batches by time requires that each job define estimated_run_minutes"
             )
 
+    def check_submission_groups(self, submitter_params):
+        """Check for invalid job submission group assignments.
+        Make a default group if none are defined and assign it to each job.
+
+        Parameters
+        ----------
+        submitter_params : SubmitterParams
+
+        Raises
+        ------
+        InvalidConfiguration
+            Raised if submission group assignments are invalid.
+
+        """
+        groups = self.submission_groups
+        if not groups:
+            self._assign_default_submission_group(submitter_params)
+            return
+
+        first_group = next(iter(groups))
+        must_be_same = ("max_nodes", "poll_interval")
+        user_overrides = ("generate_reports", "resource_monitor_interval", "dry_run", "verbose")
```

Comment:
Need more checking here. Like what happens if there are setup/shutdown scripts defined on the command line and in the submission groups. Those could be specific to the groups.